ibm/resource_ibm_is_dedicated_host.go

```diff
@@ -520,17 +520,18 @@ func resourceIbmIsDedicatedHostDelete(context context.Context, d *schema.Resourc
 		return diag.FromErr(err)
 	}
 
-	updateDedicatedHostOptions := &vpcv1.UpdateDedicatedHostOptions{}
-	dedicatedHostPrototypeMap := map[string]interface{}{}
-	dedicatedHostPrototypeMap["instance_placement_enabled"] = core.BoolPtr(false)
-	updateDedicatedHostOptions.SetID(d.Id())
-	updateDedicatedHostOptions.SetDedicatedHostPatch(dedicatedHostPrototypeMap)
-	_, updateresponse, err := vpcClient.UpdateDedicatedHostWithContext(context, updateDedicatedHostOptions)
-	if err != nil {
-		log.Printf("[DEBUG] UpdateDedicatedHostWithContext failed %s\n%s", err, updateresponse)
-		return diag.FromErr(err)
+	if *dedicatedHost.LifecycleState != isDedicatedHostSuspended && *dedicatedHost.LifecycleState != isDedicatedHostFailed {
```

Comment:
Lifecycle state cannot be suspended and failed at once..so it should be || not && condition.. and also check for nil pointers
`if dedicatedHost!=nil && dedicatedHost.LifecycleState!=nil&&(*dedicatedHost.LifecycleState != isDedicatedHostSuspended ||*dedicatedHost.LifecycleState != isDedicatedHostFailed ){`

Reply:
Will add the nil checks. With || if the state is not suspended but is failed, it will still enter the loop and do update.